### PR TITLE
Add security group subcollection to providers, cloud tenants, and vms

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

**Comment by tzumainn** [comment]
Required by https://github.com/ManageIQ/manageiq-ui-classic/pull/1997

File: app/controllers/api/subcollections/security_groups.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Api
+  module Subcollections
+    module SecurityGroups
+      def security_groups_query_resource(object)
+        object.respond_to?(:security_groups) ? object.security_groups : []
```
Comment (imtayadeway) [inline app/controllers/api/subcollections/security_groups.rb]: Can you add a test that executes the case when the object does not respond to `:security_groups`?

Comment (tzumainn) [reply app/controllers/api/subcollections/security_groups.rb]: Done - thanks!

**Comment by abellotti** [comment]
@tzumainn when you get a chance, can you add tests for accessing security_groups subcollection for vms and cloud_tenants, similar to what you did for providers ?  Thanks.

**Comment by tzumainn** [comment]
@abellotti Thanks for taking a look! I've added the tests; they depend on https://github.com/ManageIQ/manageiq/pull/16127 being merged in manageiq, though.

**Comment by abellotti** [comment]
Thanks @tzumainn for the update, I'll look at both.

**Comment by abellotti** [comment]
@tzumainn can you update the Style/Align rubocop issue and repush ?  Thanks.

**Comment by tzumainn** [comment]
@abellotti Done! (I think - I'm a bit uncertain about the frozen string comment, but I guess I'll see).

File: app/controllers/api/subcollections/security_groups.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ module Api
   module Subcollections
     module SecurityGroups
       def security_groups_query_resource(object)
-        object.respond_to?(:security_groups) ? object.security_groups : []
+        object.respond_to?(:security_groups) ? object.security_groups : [].freeze
```
Comment (abellotti) [inline app/controllers/api/subcollections/security_groups.rb]: this .freeze is not needed. can you remove it/repush ?  Thanks.  the frozen literal rubocop is unrelated.

**Comment by tzumainn** [comment]
@abellotti Ah, that's good to know. I've removed it!

**Comment by himdel** [comment]
Looks like create & update (& delete) will also be needed for https://github.com/ManageIQ/manageiq-ui-classic/pull/1997 .

**Pull request merged**